ORST/Assets/Scripts/Core/ModuleTasks/ModuleTask.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Sirenix.OdinInspector;
+using UnityEngine;
+
+namespace ORST.Core.ModuleTasks
+{
+    public enum ModuleTaskState {
+        Successful,
+        Failure,
+        Running
+    }
+
+    public class ModuleTask : MonoBehaviour {
+        [SerializeField, InlineButton(nameof(FindModuleSubtasks), "Find Subtasks")] private List<ModuleTask> m_ModuleSubtasks;
+        [SerializeField] private bool m_IsEligibleForRandom;
+
+        private Queue<ModuleTask> m_ModuleSubtaskQueue;
+        private ModuleTask m_CurrentModuleSubtask;
+
+        public bool IsEligibleForRandom() {
+            return m_IsEligibleForRandom;
+        }
```

Comment:
What do you think about replacing this method with a property? Not sure where this is used so you'll have to update those places as well if we change this.

```suggestion
        public bool IsEligibleForRandom => m_IsEligibleForRandom;
```